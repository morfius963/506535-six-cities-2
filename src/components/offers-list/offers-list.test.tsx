import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import OffersList from "./offers-list";
import OfferCard from "../offer-card/offer-card";
import Map from "../map/map";
import OffersSort from "../offers-sort/offers-sort";
import fixtureData from "../../__fixtures__/offers";

jest.mock(`../map/map`, () => jest.fn().mockReturnValue(null));
jest.mock(`../offer-card/offer-card`, () => jest.fn().mockReturnValue(null));
jest.mock(`../offers-sort/offers-sort`, () => jest.fn().mockReturnValue(null));

describe(`snapshot test`, () => {
  const appData = {
    offers: fixtureData
  };
  const user = {
    city: `Amsterdam`,
    activeSort: `Popular`
  };

  const mockStore = configureStore([]);
  const store = mockStore({
    appData,
    user
  });

  it(`Component correctly renders`, () => {
    const props = {
      activeCardId: 0,
      onFavoriteCardToggle: jest.fn(),
      onOffersSort: jest.fn()
    };

    const tree = renderer
      .create(
          <Provider store={store}>
            <OffersList {...props} />
          </Provider>
      )
      .toJSON();

    expect(Map).toHaveBeenCalled();
    expect(OfferCard).toHaveBeenCalled();
    expect(OffersSort).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
