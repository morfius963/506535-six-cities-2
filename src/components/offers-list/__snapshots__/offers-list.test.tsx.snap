// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Component correctly renders 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
        0
         places to stay in 
        Amsterdam
      </b>
      <div
        className="price-filter"
      >
        <span
          className="price-filter__value"
        >
          Choose price: 
          100€ - 3100€
        </span>
        <input
          className="price-filter__input"
          id="currentMinPrice"
          max={3100}
          min={100}
          onChange={[Function]}
          onMouseUp={[Function]}
          step="10"
          type="range"
          value={100}
        />
        <input
          className="price-filter__input"
          id="currentMaxPrice"
          max={3100}
          min={100}
          onChange={[Function]}
          onMouseUp={[Function]}
          step="10"
          type="range"
          value={3100}
        />
      </div>
      <div
        className="offer-rating"
      >
        <span>
          Choose max rating:
        </span>
        <input
          className="offer-rating__value"
          maxLength={3}
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
        <span
          className="offer-rating__description"
        >
          * input numbers from 0 to 5 to choose offer rating
        </span>
      </div>
      <div
        className="cities__places-list places__list tabs__content"
      />
    </section>
    <div
      className="cities__right-section"
    >
      <section
        className="cities__map map"
      />
    </div>
  </div>
</div>
`;
